File: python/30_set_intersection.py
import sets
import sys

with open(sys.argv[1], 'r') as fh:
    for line in fh.readlines():
        line = line.strip()
        if line == '':
            continue

        (s1, s2) = [sets.Set(s.split(',')) for s in line.split(';')]

        print ','.join(sorted(s1.intersection(s2), key = lambda v: int(v)))

File: perl/30_set_intersection.pl
#!/usr/bin/env perl

use strict;
use warnings;

open my $fh, '<', $ARGV[0];

for my $line ( <$fh> ) {
	chomp $line;
	next if $line eq '';

	my @sets = split /;/, $line;
	my %set1 = map { $_ => $_ } split /,/, $sets[0];
	my %set2 = map { $_ => $_ } split /,/, $sets[1];

	print join ',', sort { $a <=> $b } grep { $set1{$_} } keys %set2;
	print "\n";
}

close $fh;

